### ISSUE TITLE
Integrating a Non-Local Neural Block with Mask R-CNN?

### ISSUE THREAD

**Issue by cbsudux** [matterport/Mask_RCNN#679]
I'm trying to integrate a non local block (https://github.com/titu1994/keras-non-local-nets) with Mask R-CNN.

From the paper [https://arxiv.org/pdf/1711.07971.pdf](https://arxiv.org/pdf/1711.07971.pdf) "We modify the Mask R-CNN backbone by adding one non-local block (right before the last residual block of res4). All models are fine-tuned from ImageNet pre-training. We evaluate on a standard baseline of ResNet-50/101 and a high baseline of ResNeXt-152 (X152)"

How do I go about this?

**Comment by ustcjerry** [comment]
> I'm trying to integrate a non local block (https://github.com/titu1994/keras-non-local-nets) with Mask R-CNN.
> 
> From the paper https://arxiv.org/pdf/1711.07971.pdf "We modify the Mask R-CNN backbone by adding one non-local block (right before the last residual block of res4). All models are fine-tuned from ImageNet pre-training. We evaluate on a standard baseline of ResNet-50/101 and a high baseline of ResNeXt-152 (X152)"
> 
> How do I go about this?

So, have you finished this, how does the performance go ?

**Comment by cbsudux** [comment]
Nope.
